File: backend/src/main/java/com/huertohogar/model/entity/Product.java
package com.huertohogar.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Entity
@Table(name = "products")
@Data
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class Product {
    
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;
    
    @Column(nullable = false, length = 200)
    private String name;
    
    @Column(length = 2000)
    private String description;
    
    @Column(nullable = false)
    private Double price;
    
    private Double oldPrice;
    
    @Column(nullable = false)
    private Double stock;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    private ProductCategory category;
    
    @Column(name = "image_url", length = 500)
    private String imageUrl;
    
    @Column(length = 50)
    private String tag;
    
    @Column(length = 200)
    private String origin;
    
    @Column(length = 20)
    private String unit = "kg";
    
    @Column(name = "is_organic")
    private Boolean isOrganic = false;
    
    @Column(length = 500)
    private String certifications;
    
    @Column(name = "sustainable_practices", length = 500)
    private String sustainablePractices;
    
    @CreatedDate
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
    
    @LastModifiedDate
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    public enum ProductCategory {
        FRUTAS_FRESCAS,
        VERDURAS_ORGANICAS,
        PRODUCTOS_ORGANICOS,
        PRODUCTOS_LACTEOS
    }
}

File: app/src/main/java/com/huertohogar/data/model/Product.kt
package com.huertohogar.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * Categorías de productos
 */
enum class ProductCategory {
    FRUTAS_FRESCAS,
    VERDURAS_ORGANICAS,
    PRODUCTOS_ORGANICOS,
    PRODUCTOS_LACTEOS
}

/**
 * Nuestro modelo de Producto completo con toda la informacion requerida
 */
@Entity(tableName = "products")
data class Product(
    @PrimaryKey
    val id: String,
    val name: String,
    val description: String,
    val price: Double, // Precio en CLP osea en pesos chilenos
    val oldPrice: Double? = null, // Precio anterior para mostrar descuentos
    val stock: Double,
    val category: ProductCategory,
    val imageUrlName: String,
    val tag: String = "Frescos",
    val origin: String = "",
    val unit: String = "kg", // Unidad de medida (kg, bolsa, frasco, etc.)
    val isOrganic: Boolean = false, // Si es organico
    val certifications: String = "", // Certificaciones organicas
    val sustainablePractices: String = "", // Practicas sostenibles
    val isActive: Boolean = true // Si el producto está activo/pausado
)

/**
 * los Testimonios para la seccion de reseñas
 */
data class Testimonial(
    val quote: String,
    val name: String,
    val profession: String,
    val imageUrlName: String
)